### PR TITLE
feat(install): fetch recipes from recipe service

### PR DESCRIPTION
**Pull request by ctrombley**

### CODE REVIEW

File: build/lint.mk [hunk 1]
```diff
@@ -64,7 +64,7 @@ lint-commit: deps
 
 golangci: deps
 	@echo "=== $(PROJECT_NAME) === [ golangci-lint    ]: Linting using $(GOLINTER) ($(COMMIT_LINT_CMD))..."
-	@$(GOLINTER) run
+	@$(GOLINTER) run --build-tags unit,integration
```
Comment (ctrombley) [inline build/lint.mk:67]: This is necessary to make the linter look at our test files.

File: internal/install/command.go [hunk 2]
```diff
@@ -12,8 +12,6 @@ import (
 	"github.com/newrelic/newrelic-client-go/newrelic"
 )
 
-var configFiles []string
```
Comment (ctrombley) [inline internal/install/command.go:15]: Let's pull this out and eventually replace it with a recipe search parameter to be forwarded to the service.

File: internal/install/command.go [hunk 3]
```diff
@@ -25,15 +23,11 @@ var Command = &cobra.Command{
 				log.Fatal(errors.New("default profile has not been set"))
 			}
 
-			if err := install(configFiles); err != nil {
+			if err := install(nrClient); err != nil {
```
Comment (ctrombley) [inline internal/install/command.go:26]: Using DI to inject a client instance.

File: internal/install/discoverer.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package install
+
+import "context"
+
+type discoverer interface {
```
Comment (ctrombley) [inline internal/install/discoverer.go:5]: Refactored this file around the idea of a `discoverer` interface and related concepts.

File: internal/install/discoverer.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package install
```
Comment (ctrombley) [inline internal/install/discoverer.go:1]: Refactored this file around the idea of a `discoverer` interface and related concepts.

File: internal/install/install.go [hunk 6]
```diff
@@ -17,30 +17,34 @@ import (
 	"gopkg.in/yaml.v2"
 
 	"github.com/newrelic/newrelic-cli/internal/credentials"
+	"github.com/newrelic/newrelic-client-go/newrelic"
 )
 
-func install(configFiles []string) error {
+func install(client *newrelic.NewRelic) error {
+	ctx := getSignalContext()
```
Comment (ctrombley) [inline internal/install/install.go:24]: Hoist the signal-capable context up to the top of the `install` method and use it for all IO-blocking operations.

File: internal/install/install.go [hunk 7]
```diff
@@ -17,30 +17,34 @@ import (
...
+	rf := newServiceRecipeFetcher(&client.NerdGraph)
```
Comment (ctrombley) [inline internal/install/install.go:25]: Using DI to inject a nerdgraph client instance into a `serviceRecipeFetcher` instance.

File: internal/install/install.go [hunk 8]
```diff
@@ -17,30 +17,34 @@ import (
...
+	pf := newRegexProcessFilterer(rf)
```
Comment (ctrombley) [inline internal/install/install.go:26]: Using DI to inject a `serviceRecipeFetcher` instance into a `regexProcessFilterer` instance.

File: internal/install/install.go [hunk 9]
```diff
@@ -17,30 +17,34 @@ import (
...
+
 	// Execute the discovery process.
 	log.Debug("Running discovery...")
-	var d discoverer = new(psUtilDiscoverer)
-	manifest, err := d.discover()
+	var d discoverer = newPSUtilDiscoverer(pf)
```
Comment (ctrombley) [inline internal/install/install.go:30]: Using DI to inject a `regexProcessFilterer` instance into a `psUtilDiscoverer` instance.

File: internal/install/install.go [hunk 10]
```diff
@@ -17,30 +17,34 @@ import (
...
 		return err
 	}
 
-	log.Debugf("manifest: %+v", manifest)
-
 	// Retrieve the relevant recipes.
 	log.Debug("Retrieving recipes...")
-	var f recipeFetcher = new(yamlRecipeFetcher)
-	recipes, err := f.fetch(configFiles, manifest)
+	// TODO: pass context into this method.  The client will need to be updated
```
Comment (ctrombley) [inline internal/install/install.go:38]: We will want to pass context into the fetch methods here, but the underlying client does not yet support it.

File: internal/install/mock_nerdgraph_client.go [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package install
```
Comment (ctrombley) [inline internal/install/mock_nerdgraph_client.go:1]: A mock nerdgraph client for unit tests.

File: internal/install/mock_process.go [hunk 12]
```diff
@@ -0,0 +1,14 @@
+package install
```
Comment (ctrombley) [inline internal/install/mock_process.go:1]: A mock process struct for unit tests.

File: internal/install/mock_recipe_fetcher.go [hunk 13]
```diff
@@ -0,0 +1,30 @@
+package install
```
Comment (ctrombley) [inline internal/install/mock_recipe_fetcher.go:1]: A mock recipe fetcher for unit tests.

File: internal/install/nerdgraph_client.go [hunk 14]
```diff
@@ -0,0 +1,5 @@
+package install
```
Comment (ctrombley) [inline internal/install/nerdgraph_client.go:1]: An interface that matches the nerdgraph client exported from `newrelic-client-go` to facilitate unit testing.

**Review by zlesnr — APPROVED** [review]

File: internal/install/discoverer.go [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+	kernelArch      string
+}
+
+type genericProcess interface {
+	Name() (string, error)
+	PID() int32
+}
+
+func (d *discoveryManifest) AddProcess(p genericProcess) {
+	d.processes = append(d.processes, p)
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply internal/install/discoverer.go:25]: Will leave as-is per conversation.

**Pull request merged**